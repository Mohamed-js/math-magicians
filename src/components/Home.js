import React from 'react';

const Home = () => (
  <>
    <h1>Math Magicians</h1>
    <h3>All magic happens here!</h3>
  </>
);

export default Home;
