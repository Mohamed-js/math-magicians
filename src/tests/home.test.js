import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../components/Home';

// Test rendering Home components
test('renders Home header', () => {
  const { getByText } = render(<Home />);
  const headerText = getByText(/Math Magicians/);
  expect(headerText).toBeInTheDocument();
});

test('renders Home slogan', () => {
  const { getByText } = render(<Home />);
  const headerText = getByText(/Where all the magic happens!/);
  expect(headerText).toBeInTheDocument();
});
