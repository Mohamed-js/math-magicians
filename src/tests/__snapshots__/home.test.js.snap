// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About link renders correctly 1`] = `"<Link to=\\"/about\\">About</Link>"`;

exports[`Calculator link renders correctly 1`] = `"<Link to=\\"/calculator\\">Calculator</Link>"`;

exports[`Calculator link renders correctly 2`] = `"<Home />"`;

exports[`Home link renders correctly 1`] = `"<Link to=\\"/\\">Home</Link>"`;
